docs/enhancements/boot-status-reporter.md

```diff
@@ -0,0 +1,129 @@
+---
+title: boot-status-reporter
+authors:
+  - "@nmagnezi"
+creation-date: 2022-09-22
+last-updated: 2022-09-22
+---
+
+# Assisted Boot Reporter
+
+## Summary
+
+Create a new reporting service that runs as early as possible when hosts are booted from disk to collect additional 
+information and send it to `assisted-service`.
+
+## Motivation
+
+There are multiple occurrences of clusters that fail to install at the stage where
+hosts manage to pull ignition but fail to get to a point where the `assisted-controller`
```

Comment:
I think one motivation for this service (in case we intend to use it for more than just reporting successful boot) is to align the SNO stages and logs gathering functionality with the multi-node stages.

Reply:
What scenario is this?
Why would it apply the new boot-reporter service but not the rest of the ignition payload?

Reply:
Agree with Igal here. We had cases where we managed to pull the ignition but RHCOS failed to apply the ignition. E.g. using `storage` stanza that refers to a disk via non-existing name. That's syntax-wise a totally correct ignition override but RHCOS will refuse to apply it and will not continue the boot process.

Whether this service here would help is another question as we would need to design its dependency chain quite carefully.